pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -379,7 +424,11 @@ public Response stopFunctionInstances(String tenant, String namespace, String fu
      * It stops all functions instances owned by current worker
      * @throws Exception
      */
-    public void stopAllOwnedFunctions() throws Exception {
+    public void stopAllOwnedFunctions() {
+        if (runtimeFactory.externallyManaged()) {
```

Comment:
why not stop external functions here? So what happens when the worker that the external function is assigned to dies?  It seems with the current logic in the checkFailures method in MembershipManager,  the external function will get reassigned and restarted.  Is that what we want? You might also get a name conflict in kubernetes when the function gets restarted since the function already exists in kubernetes.

Reply:
this function is called during exit of the worker(either unplanned or planned). At this point we don't really want the functions to die. Thus this check.
When the function reassignment happens, a new worker will get this assignment, and try starting the service, which will result in conflict which is then treated as function running by the k8 runtime. Thus the function is not started and yet the worker will have the right notion that it exists